File: Easy/Python/1800. Maximum Ascending Subarray Sum.py
class Solution(object):
    def maxAscendingSum(self, nums):
        this_sum = nums[0]
        max_sum = this_sum

        for i in range(1, len(nums)):
            if nums[i - 1] < nums[i]:
                this_sum += nums[i]
            else:
                max_sum = max(max_sum, this_sum)
                this_sum = nums[i]
        
        return this_sum if max_sum < this_sum else max_sum
        

File: Easy/Java/1800. Maximum Ascending Subarray Sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int sum = nums[0];
        int maxSum = sum;

        for(int i = 1; i < nums.length; i++){
            if(nums[i - 1] < nums[i]){
                sum += nums[i];
            }else{
                maxSum = Math.max(maxSum, sum);
                sum = nums[i];
            }
            
        }

        return maxSum < sum ? sum : maxSum;
    }
}
